components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -421,121 +418,98 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
         return [];
     }
 
-    public async getWorkspace(id: string): Promise<WorkspaceInfo> {
-        const user = this.checkUser('getWorkspace');
-        const span = opentracing.globalTracer().startSpan("getWorkspace");
-        span.setTag("workspaceId", id);
-        span.setTag("userId", user.id);
+    public async getWorkspace(ctx: TraceContext, workspaceId: string): Promise<WorkspaceInfo> {
+        // TODO(gpl) Add all parameters as tags generically?
```

Comment:
interesting idea! usually this is something we receive over WS connection, so it should be safe to add. though, maybe we can add a sanitizer and introduce some length-based filters, wdyt?

Reply:
> introduce some length-based filters

Yep, I think that would help. Maybe opentracing already has sth like this. Looking forward to @mads-hartmann sharing his experience a bit.

> sanitizer

Do you have sth concrete in mind? :thinking:

Reply:
we cannot weight the params objects before serializing, and an additional serializing step seems quite expensive, right?

Reply:
> additional serializing step seems quite expensive

:100: my thoughts.

But as Mads opts for "add everthing generically" I'll go with that.